### ISSUE TITLE
Add templater for snowsql

### ISSUE THREAD

**Issue by chwiese** [sqlfluff/sqlfluff#1395]
Snowsql, the official cli client for Snowflake, [has support for variables](https://docs.snowflake.com/en/user-guide/snowsql-use.html#using-variables). Currently the use of these variables results in a parsing error, e.g. `USE WAREHOUSE &warehouse;` results in
```
L:   5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'USE
                       | WAREHOUSE &warehouse'
```

As far as I understand the sqlfluff architecture, a (very simple?) templater for snowsql might address this.

**Comment by tunetheweb** [comment]
SQLFluff already supports the `USE WAREHOUSE` syntax:

```
echo "USE WAREHOUSE warehouse;" | sqlfluff parse --dialect snowflake -
```

However it doesn't support the `&warehouse` variable syntax.

Similar variable support was added to BigQuery recently in #1363 so should be easy enough to do similar here if you (or anyone else) fancies working on this?

**Comment by tunetheweb** [comment]
@samlader i presume this  can be closed now? Or is there more to do? Will go ahead and close but feel free to reopen if there’s still more to be done.

FYI if you say “Fixes #1395” instead of “Makes Progress on #1395” in your PR then GitHub will automatically close the issue when it’s merged.
